mmyolo/datasets/transforms/transforms.py

```diff
@@ -558,32 +630,143 @@ def transform(self, results: dict) -> dict:
             borderValue=self.border_val)
         results['img'] = img
         results['img_shape'] = img.shape
+        img_h, img_w = img.shape[:2]
 
         bboxes = results['gt_bboxes']
         num_bboxes = len(bboxes)
         if num_bboxes:
             orig_bboxes = bboxes.clone()
+            if self.use_mask_refine and 'gt_masks' in results:
+                # If the dataset has annotations of mask,
+                # the mask will be used to refine bbox.
+                gt_masks = results['gt_masks']
+
+                gt_masks_resample = self.resample_masks(gt_masks)
+                gt_masks = self.warp_mask(gt_masks_resample, warp_matrix,
+                                          img_h, img_w)
+
+                # refine bboxes by masks
+                bboxes = gt_masks.get_bboxes(dst_type='hbox')
+                # filter bboxes outside image
+                valid_index = self.filter_gt_bboxes(orig_bboxes,
+                                                    bboxes).numpy()
+                results['gt_masks'] = gt_masks[valid_index]
+            else:
+                bboxes.project_(warp_matrix)
+                if self.bbox_clip_border:
+                    bboxes.clip_([height, width])
+
+                # filter bboxes
+                orig_bboxes.rescale_([scaling_ratio, scaling_ratio])
+
+                # Be careful: valid_index must convert to numpy,
+                # otherwise it will raise out of bounds when len(valid_index)=1
+                valid_index = self.filter_gt_bboxes(orig_bboxes,
+                                                    bboxes).numpy()
+                if 'gt_masks' in results:
+                    results['gt_masks'] = PolygonMasks(
+                        results['gt_masks'].masks, img_h, img_w)
 
-            bboxes.project_(warp_matrix)
-            if self.bbox_clip_border:
-                bboxes.clip_([height, width])
-
-            # filter bboxes
-            orig_bboxes.rescale_([scaling_ratio, scaling_ratio])
-
-            # Be careful: valid_index must convert to numpy,
-            # otherwise it will raise out of bounds when len(valid_index)=1
-            valid_index = self.filter_gt_bboxes(orig_bboxes, bboxes).numpy()
             results['gt_bboxes'] = bboxes[valid_index]
             results['gt_bboxes_labels'] = results['gt_bboxes_labels'][
                 valid_index]
             results['gt_ignore_flags'] = results['gt_ignore_flags'][
                 valid_index]
 
-            if 'gt_masks' in results:
-                raise NotImplementedError('RandomAffine only supports bbox.')
         return results
 
+    @staticmethod
+    def warp_poly(poly: np.ndarray, warp_matrix: np.ndarray, img_w: int,
```

Comment:
You can refer to `bbox_project` in `mmdet/structures/bbox/transforms.py` to write a `mask_project` function.

Reply:
For quick development purposes, this function will remain the current implementation for now. I will give PR to mmdet to implement the relevant function